src/effects/RealtimeEffectState.cpp

```diff
@@ -185,14 +189,14 @@ struct RealtimeEffectState::Access final : EffectSettingsAccess {
    void Flush() override {
       if (auto pState = mwState.lock()) {
          if (auto pAccessState = pState->GetAccessState()) {
-            auto &lastSettings = pAccessState->mLastSettings;
             const EffectSettings *pResult{};
             while (!(pResult = FlushAttempt(*pAccessState))) {
                // Wait for progress of audio thread
                using namespace std::chrono;
                std::this_thread::sleep_for(50ms);
             }
             pState->mMainSettings.Set(*pResult); // Update the state
+            pAccessState->mLastSettings = *pResult;
```

Comment:
Thanks for finding the error in my Access class.  I think I suggested making the fix in `FlushAttempt` instead.  That might update the main-thread version of output port values a little more frequently, maybe improving update rate seen in theater?  But either way is sufficient for `Flush().`

Reply:
After testing the code it turned out, that it results in problems when changing the settings from the effect UI, which you have mentioned to be the case for AU previously.

Reply:
Probably it will be fixed when CopySettings will be fixed according to your proposal as well